Userland/Libraries/LibTextCodec/Decoder.cpp

```diff
@@ -171,38 +171,43 @@ Decoder* bom_sniff_to_decoder(StringView input)
 
     auto bytes = input.bytes();
     if (bytes.size() < 2)
-        return nullptr;
+        return {};
 
     auto first_byte = bytes[0];
 
     switch (first_byte) {
     case 0xEF: // UTF-8
         if (bytes.size() < 3)
-            return nullptr;
-        return bytes[1] == 0xBB && bytes[2] == 0xBF ? &s_utf8_decoder : nullptr;
+            return {};
+        if (bytes[1] == 0xBB && bytes[2] == 0xBF)
+            return s_utf8_decoder;
+        return {};
     case 0xFE: // UTF-16BE
-        return bytes[1] == 0xFF ? &s_utf16be_decoder : nullptr;
+        if (bytes[1] == 0xFF)
+            return s_utf16be_decoder;
+        return {};
     case 0xFF: // UTF-16LE
-        return bytes[1] == 0xFE ? &s_utf16le_decoder : nullptr;
+        if (bytes[1] == 0xFE)
+            return s_utf16le_decoder;
+        return {};
     }
 
-    return nullptr;
+    return {};
 }
 
 // https://encoding.spec.whatwg.org/#decode
-DeprecatedString convert_input_to_utf8_using_given_decoder_unless_there_is_a_byte_order_mark(Decoder& fallback_decoder, StringView input)
+ErrorOr<String> convert_input_to_utf8_using_given_decoder_unless_there_is_a_byte_order_mark(Decoder& fallback_decoder, StringView input)
 {
     Decoder* actual_decoder = &fallback_decoder;
 
     // 1. Let BOMEncoding be the result of BOM sniffing ioQueue.
     // 2. If BOMEncoding is non-null:
-    if (auto* unicode_decoder = bom_sniff_to_decoder(input); unicode_decoder) {
+    if (auto unicode_decoder = bom_sniff_to_decoder(input); unicode_decoder.has_value()) {
         // 1. Set encoding to BOMEncoding.
-        actual_decoder = unicode_decoder;
+        actual_decoder = &unicode_decoder.value();
 
         // 2. Read three bytes from ioQueue, if BOMEncoding is UTF-8; otherwise read two bytes. (Do nothing with those bytes.)
-        // FIXME: I imagine this will be pretty slow for large inputs, as it's regenerating the input without the first 2/3 bytes.
```

Comment:
Keep fix me?

Reply:
Why is getting a smaller StringView from a StringView a slow operation though?

Reply:
I think it tries to say that it makes a large copy when returning, not that this line here is slow (?)

(The general point is "don't make unrelated changes" – this PR isn't about this FIXME, so it shouldn't touch it imho.)

Reply:
I'll drop the commit, it seems silly to hold up the rest of the PR over this. :^)